src/memory.rs

```diff
@@ -887,6 +902,12 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
         Ok(self.get(ptr.alloc_id)?.relocations.range(start..end))
     }
 
+    fn has_non_int_relocations(&self, ptr: Pointer, size: u64)
+        -> EvalResult<'tcx, bool>
+    {
+        Ok(self.relocations(ptr, size)?.any(|(_, &alloc_id)| alloc_id != NEVER_ALLOC_ID))
```

Comment:
I think it would be better to stop writing int relocations to the relocation table.